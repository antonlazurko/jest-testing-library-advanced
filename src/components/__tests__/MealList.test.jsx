import { screen } from "@testing-library/react";
import { renderWithRouter } from "../../utils/testing";
import { MealList } from "../MealList";

describe("MealList", () => {
    test("should render correctly", () => {
        renderWithRouter(<MealList meals={[{
            idMeal: "test",
            strMeal: "test",
            strMealThumb: "/test.png"
        }, {idMeal: "test2", strMeal: "test", strMealThumb: "/test.png"}]}/>);
        expect(screen.getByRole("list")).toMatchSnapshot();
    });
})