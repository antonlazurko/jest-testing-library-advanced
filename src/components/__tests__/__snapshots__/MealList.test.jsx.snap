// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealList should render correctly 1`] = `
<div
  class="list"
  role="list"
>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="test"
        src="/test.png"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        test
      </span>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/meal/test"
      >
        Watch recipe
      </a>
    </div>
  </div>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="test"
        src="/test.png"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        test
      </span>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/meal/test2"
      >
        Watch recipe
      </a>
    </div>
  </div>
</div>
`;
